### PR TITLE
Thing update event

### PR DESCRIPTION
**Pull request by aounhaider1**
fixes https://github.com/eclipse/smarthome/issues/2091
Signed-off-by: Aoun Bukhari <bukhari@itemis.de>

### CODE REVIEW

**Comment by maggu2810** [comment]
@sjka As you created #2091, can you check that changes, too?

**Comment by sjka** [comment]
The `Thing` itself now gets properly updated, e.g. label changes are reflected instantly. Nice!

However, not its channels. Let's assume you have the following:

```
yahooweather:weather:losangeles "Los Angeles 1" [ location=2442047, refresh=60 ] {
    Channels: 
        String : foo
}
```

Navigate to Configuration -> Things -> Los Angeles 1. You should see the three original channels plus the "foo" one. Now change the file to the following:

```
yahooweather:weather:losangeles "Los Angeles 1" [ location=2442047, refresh=60 ] {
    Channels: 
        String : foo
        String : bar
}
```

The "bar" channel will only show up after a reload of the page.

**Comment by aounhaider1** [comment]
Updated accordingly! However, if you are editing a channel's configurations in two browsers and save one of them, the change will not be directly visible unless you close and reopen the dialog. Implementing it should be possible but I am not sure of it is worth the effort.

**Comment by aounhaider1** [comment]
Needs updating.

**Comment by aounhaider1** [comment]
It should now support adding/removing channels.

File: extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.configuration.js [hunk 1]
```diff
@@ -662,7 +666,7 @@ angular.module('PaperUI.controllers.configuration', [ 'PaperUI.constants' ]).con
         thingService.update({
             thingUID : thing.UID
         }, $scope.thing, function() {
-            thingRepository.update($scope.thing);
+            // thingRepository.update($scope.thing);
```
Comment (sjka) [inline extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.configuration.js]: if that's not needed here anymore, could you please delete it?

**Comment by sjka** [comment]
thanks. I only found a left-over commented-out code line. Could you please remove it? 
Otherwise lgtm

**Comment by aounhaider1** [comment]
Updated.

**Pull request merged**